datasets/librispeech_asr/librispeech_asr.py

```diff
@@ -117,61 +133,101 @@ def _info(self):
-            datasets.SplitGenerator(
-                name=datasets.Split.VALIDATION, gen_kwargs={"files": dl_manager.iter_archive(archive_path["dev"])}
-            ),
-            datasets.SplitGenerator(
-                name=datasets.Split.TEST, gen_kwargs={"files": dl_manager.iter_archive(archive_path["test"])}
-            ),
-        ]
+        return train_splits + dev_splits + test_splits
 
     def _generate_examples(self, files):
         """Generate examples from a LibriSpeech archive_path."""
         key = 0
         audio_data = {}
         transcripts = []
-        for path, f in files:
-            if path.endswith(".flac"):
-                id_ = path.split("/")[-1][: -len(".flac")]
-                audio_data[id_] = f.read()
-            elif path.endswith(".trans.txt"):
-                for line in f:
-                    if line:
-                        line = line.decode("utf-8").strip()
-                        id_, transcript = line.split(" ", 1)
-                        audio_file = f"{id_}.flac"
-                        speaker_id, chapter_id = [int(el) for el in id_.split("-")[:2]]
-                        transcripts.append(
-                            {
-                                "id": id_,
-                                "speaker_id": speaker_id,
-                                "chapter_id": chapter_id,
-                                "file": audio_file,
-                                "text": transcript,
-                            }
-                        )
-            if audio_data and len(audio_data) == len(transcripts):
-                for transcript in transcripts:
-                    audio = {"path": transcript["file"], "bytes": audio_data[transcript["id"]]}
-                    yield key, {"audio": audio, **transcript}
-                    key += 1
-                audio_data = {}
-                transcripts = []
+        for subset, iterator in files.items():
+            for path, f in iterator:
+                if path.endswith(".flac"):
+                    id_ = path.split("/")[-1][: -len(".flac")]
+                    audio_data[id_] = f.read()
+                elif path.endswith(".trans.txt"):
+                    for line in f:
+                        if line:
+                            line = line.decode("utf-8").strip()
+                            id_, transcript = line.split(" ", 1)
+                            audio_file = f"{id_}.flac"
+                            speaker_id, chapter_id = [int(el) for el in id_.split("-")[:2]]
+                            transcripts.append(
+                                {
+                                    "id": id_,
+                                    "speaker_id": speaker_id,
+                                    "chapter_id": chapter_id,
+                                    "file": audio_file,
+                                    "text": transcript,
+                                    "subset": subset,
```

Comment:
@albertz @lhoestq - this "subset" arg is one of "clean-100", "clean-360", "other-500" for:

```python
ds = load_dataset("librispeech_asr", "all", split="train")
ds[0]
```

This should allow the user to easily define which subsets are needed I think